### ISSUE TITLE
Offsets in Compose break on with RNG

### ISSUE THREAD

**Issue by yschimke** [robolectric/robolectric#8592]
### Description

A LazyColumn with items with offsets applied via graphics modifiers, breaks the RNG somehow.  

Notice the truncation of the top of the chips and known on effect of uneven gaps.

### Steps to Reproduce

Composable code

```
@Composable
fun MyTest() {
    val items = listOf(
        Data(0.91855353f, 0.9511321f, 0f),
        Data(1.0f, 1.0f, 0f),
        Data(0.8984337f, 0.8307228f, 4f),
        Data(0.74426603f, 0.5737767f, -8f)
    )
    LazyColumn {
        items(items) {
            ScalingLazyColumnItemWrapper(currentItem = it)
        }
    }
}

data class Data(
    val alpha: Float,
    val scale: Float,
    val offsetAdjust: Float,
)

@Composable
private fun ScalingLazyColumnItemWrapper(
    currentItem: Data
) {
    Box(
        modifier = Modifier.graphicsLayer {
            alpha = currentItem.alpha
            scaleX = currentItem.scale
            scaleY = currentItem.scale
            if (currentItem.scale > 0f) {
                translationY = currentItem.offsetAdjust
                transformOrigin = TransformOrigin(
                    pivotFractionX = 0.5f,
                    pivotFractionY = 0.0f
                )
            }
        }
    ) {
        Chip(label = "Item", onClick = { /*TODO*/ })
    }
}
```

Emulator results 

![image](https://github.com/robolectric/robolectric/assets/231923/0c420c44-1601-408e-a3f5-fe2be0216d5b)

Robolectric (via screenshot testing)

![image](https://github.com/robolectric/robolectric/assets/231923/d5203563-bbfd-4898-a840-c2c2e429c922)

Compose Previews

![image](https://github.com/robolectric/robolectric/assets/231923/27f9c369-e32b-4aee-a51b-cfa1bf25677e)

### Robolectric & Android Version

Robolectric: 4.10.3
SDK : 30

**Comment by yschimke** [comment]
There are in place tests in https://github.com/google/horologist/pull/1792 that reproduce the problem, but are harder to debug.

**Comment by hoisie** [comment]
We can take a look at this, thanks for making a concise repro 👍 

Note that while we want RNG to have as high-fideilty rendering as possible, if you really want pixel-perfect screenshot testing, our recommendation would always be to use emulator-based tests. That being said, if the result looks fine in LayoutLib, maybe there is something RNG could do better.

**Comment by hoisie** [comment]
Where does 'Chip' come from? Is this from the horologist project? Is it possible to reproduce with a more basic/built-in component?

**Comment by yschimke** [comment]
Yep. Will change my repro to use Wear Compose. It's a minimal change.

**Comment by yschimke** [comment]
Here is a repro in a single file https://github.com/yschimke/horologist/blob/15a58a7c98b3ce894a2596b18032cd4ee810d0a1/sample/src/main/java/com/google/android/horologist/scratch/ScratchActivity.kt

It doesn't have screenshot tests, but has working composable code.

If you want an example with something like roborazzi, let me know, but I will have to make it from scratch.

**Comment by hoisie** [comment]
We did some initial investigation and the problem appears to be related to HW rendering. When HW rendering is used, it appears to render fine. Maybe there is some clipping-to-outline being used underneath the hood, which does not work with SW rendering. We have HW rendering working in RNG but are ironing out some issues before a public release.

See https://android.googlesource.com/platform/frameworks/base/+log/refs/heads/android12-hostruntime-dev for commits.

**Comment by yschimke** [comment]
Can I force compose on device/emulator/previews to SW rendering also? To confirm that?

If so, is that a compose bug?

**Comment by yschimke** [comment]
The code for native rendering

https://cs.android.com/android/platform/superproject/main/+/main:external/robolectric/shadows/framework/src/main/java/org/robolectric/shadows/ShadowPixelCopy.java;l=285?q=robolectric.screenshot.hwrdr.native

gated by

```
  private static final String USE_HARDWARE_RENDERER_NATIVE_ENV =
      "robolectric.screenshot.hwrdr.native";
```

**Comment by yschimke** [comment]
Closing this out as it's presumably a software rendering issue, and there is code to fix the tests.

**Comment by yschimke** [comment]
Reopening until it makes the release

**Comment by yschimke** [comment]
Can confirm with the code from cs.android.com with HW support it's working.

<img width="1178" alt="image" src="https://github.com/robolectric/robolectric/assets/231923/786ef3f6-38eb-45b0-b212-4d6c0efda8d2">

**Comment by hoisie** [comment]
We are still iterating a bit on the HW rendering, but we will make a point release when it is more stable.

**Comment by yschimke** [comment]
Ooops that other PR, doesn't close this.

**Comment by desugar-64** [comment]
@hoisie Correct me if I'm mistaken, but this issue should be addressed in the upcoming release of Robolectric 4.12, right? I'm experiencing a similar problem with software rendering in tests involving Jetpack Compose. 

It clips the content of a composable widget if it's visually translated beyond the widget's boundaries. To resolve this, I had to implement some workarounds in the project code to ensure proper functionality in snapshot tests with Roborazzi.
